src/SqlAst/AvgReturnTypeExtension.php

```diff
@@ -25,7 +28,23 @@ public function getReturnType(FunctionCall $expression, QueryScope $scope): ?Typ
         }
 
         $argType = $scope->getType($args[0]);
-        $numberType = $argType->toNumber();
+
+        if ($argType->isNull()->yes()) {
+            return $argType;
+        }
+        $argType = TypeCombinator::removeNull($argType);
+
+        if ($argType->isInteger()->yes()) {
+            $numberType = new IntersectionType([
+                new StringType(),
+                new AccessoryNumericStringType(),
+            ]);
+        } elseif ($argType->isSuperTypeOf(new StringType())->yes()) {
```

Comment:
isString()->yes()

Reply:
I believe that `isSuperTypeOf` is necessary here as well. For example, consider `SELECT avg(ifnull(email, adaid)) as avg from ada`. The argument to `avg` is `string|int`, but PDO will always output the result as a float (even if all `email` are null).

Reply:
Actually, I think the issue with my example above is due to the behavior of `ifnull`, and has nothing to do with avg/min/max. Let me fix this PR and I'll submit a separate PR for ifnull.

The underlying issue was:
> The default return type of [IFNULL(expr1,expr2)](https://dev.mysql.com/doc/refman/8.0/en/flow-control-functions.html#function_ifnull) is the more “general” of the two expressions, in the order STRING, REAL, or INTEGER